llvm/lib/IR/RepoDefinition.cpp

```diff
@@ -263,18 +255,19 @@ auto ModuleHashGenerator::updateDigestUseDependenciesAndContributions(
 
   GOHash.update(static_cast<char>(Tags::GO));
   auto State = accumulateGODigest(GO, GOHash, UseRepoDefinitionMD);
-  // The hashes of the GO's 'contributions' and 'dependencies' are separately
-  // added to the GO's hash since their respective arrows point are in different
-  // direction. In addition, 'contributions' cannot form loops since 1) only
-  // global variables have 'contributions' and 2) all 'contributions' are
-  // functions. Therefore, the 'contributions' can be simplified not to record
-  // 'visited' map, which avoids having two 'visited' maps. Firstly, add the
-  // GO's contributions to its hash.
-  updateDigestUseContributions(GOHash, State.Contributions);
-
-  // Add the GO's dependencies to its hash.
-  return updateDigestUseDependencies(GO, GOHash, GODepth, State.Dependencies,
-                                     UseRepoDefinitionMD);
+  // The hashes of the GO's 'contributions' and 'dependencies' are together
+  // added to the GO's hash. Although their respective arrows point are in
+  // different direction, they can affect each other and form loops.
+  // Consider the following example:
+  // int var = 0;
+  // int Func() {
+  //    var = var + 5;
+  //    return var;
+  //  }
+  // 'Func' is a contribution of 'var' and 'var' is a dependency of 'Func`.
+  // Their hash values are dependent on each other. They can form loops and must
+  // be treated equally.
```

Comment:
This comment, whilst useful doesn't really seem to relate to the line of code that it preceeds.

Reply:
Thanks, I will move the comment to function declaration.